src/nvim/api/private/defs.h

```diff
@@ -30,10 +30,14 @@ typedef enum {
   kMessageTypeNotification
 } MessageType;
 
-/// Used as the message ID of notifications.
+/// Used as response id of notifications. Api methods are allowed to not
+/// calculate the return value, as it will be discarded anyway.
 #define NO_RESPONSE UINT64_MAX
 
-/// Used as channel_id when the call is local
+/// Used as response id when the call is local or part of a multi-request.
+#define INTERNAL_RESPONSE (UINT64_MAX-1u)
```

Comment:
Actually is is not possible for a API method to receive the `response_id` (like it is with the `channel_id`), it is just an always unused argument to `handle_nvim_[method]`. I'll just remove it for now.